app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Admin::VideosController < ApplicationController
+  before_action :require_user
+  before_action :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    if @video.save
+      redirect_to new_admin_video_path
+      flash[:success] = "You have added #{@video.title}"
+    else
+      flash[:danger] = "Video has not been added - Please check the errors"
```

Comment:
Whenever you `render`, you need to use `flash.now`. `flash` will survive the next request as well.